lib/Dialect/Lattigo/Transforms/ConfigureCryptoContext.cpp

```diff
@@ -282,17 +282,51 @@ struct ConfigureCryptoContext
     : impl::ConfigureCryptoContextBase<ConfigureCryptoContext> {
   using ConfigureCryptoContextBase::ConfigureCryptoContextBase;
 
-  void runOnOperation() override {
-    auto result =
-        getOperation()->walk<WalkOrder::PreOrder>([&](func::FuncOp op) {
-          auto funcName = op.getSymName();
-          if ((funcName == entryFunction) && failed(convertFunc(op))) {
-            op->emitError("Failed to configure the crypto context for func");
-            return WalkResult::interrupt();
-          }
+  func::FuncOp getEntryFunction() {
+    ModuleOp moduleOp = cast<ModuleOp>(getOperation());
```

Comment:
could you please factor some of this logic into a utility that can be used in the other pass pipelines please?

Reply:
Put it in `lib/Utils/TransformUtils.h` now